frontend/src/components/ZetaneDrawflowEditor.js

```diff
@@ -712,322 +718,294 @@ export default class Drawflow {
+          var line_y = outputElement.offsetHeight / 2 + (outputElement.getBoundingClientRect().y - precanvas.getBoundingClientRect().y) * precanvasHeightZoom;
+  
+          var x = eX;
+          var y = eY;
+  
+          const lineCurve = createCurvature(line_x, line_y, x, y, curvature, 'openclose');
+          elemsOut[item].children[0].setAttributeNS(null, 'd', lineCurve);
+  
+        } else {
+          const points = elemsOut[item].querySelectorAll('.point');
+          let linecurve = '';
+          const reoute_fix = [];
+  
+          points.forEach((item, i) => {
+            const svgName = '.' + item.parentElement.classList.value.split(" ").join(".");
+            const { input_class, input_id, output_class, output_id } = connection_list[svgName];
+            const circleElement = item;
+            if (i === 0 && ((points.length - 1) === 0)) {
+              inputElement = nodeRefs[`${input_id}-input-${input_class}`];
+              outputElement = nodeRefs[`${output_id}-output-${output_class}`];
+  
+              var eX = (circleElement.getBoundingClientRect().x - precanvas.getBoundingClientRect().x) * precanvasWitdhZoom + rerouteWidth;
+              var eY = (circleElement.getBoundingClientRect().y - precanvas.getBoundingClientRect().y) * precanvasHeightZoom + rerouteWidth;
+  
+              var line_x = outputElement.offsetWidth / 2 + (outputElement.getBoundingClientRect().x - precanvas.getBoundingClientRect().x) * precanvasWitdhZoom;
+              var line_y = outputElement.offsetHeight / 2 + (outputElement.getBoundingClientRect().y - precanvas.getBoundingClientRect().y) * precanvasHeightZoom;
+              
+              var x = eX;
+              var y = eY;
+  
+              var lineCurveSearch = createCurvature(line_x, line_y, x, y, reroute_curvature_start_end, 'open');
+              linecurve += lineCurveSearch;
+              reoute_fix.push(lineCurveSearch);
+  
+              var eX = inputElement.offsetWidth / 2 + (inputElement.getBoundingClientRect().x - precanvas.getBoundingClientRect().x) * precanvasWitdhZoom;
+              var eY = inputElement.offsetHeight / 2 + (inputElement.getBoundingClientRect().y - precanvas.getBoundingClientRect().y) * precanvasHeightZoom;
+  
+              var line_x = (circleElement.getBoundingClientRect().x - precanvas.getBoundingClientRect().x) * precanvasWitdhZoom + rerouteWidth;
+              var line_y = (circleElement.getBoundingClientRect().y - precanvas.getBoundingClientRect().y) * precanvasHeightZoom + rerouteWidth;
+              var x = eX;
+              var y = eY;
+  
+              var lineCurveSearch = createCurvature(line_x, line_y, x, y, reroute_curvature_start_end, 'close');
+              linecurve += lineCurveSearch;
+              reoute_fix.push(lineCurveSearch);
+  
+            } else if (i === 0) { 
+              outputElement = nodeRefs[`${output_id}-output-${output_class}`];
```

Comment:
null check and log. maybe since there's a bunch of these, pull them up out of the if / else statement so you only have to null check them once?